diff --git a/cpp/src/dave/frame_processors.cpp b/cpp/src/dave/frame_processors.cpp
index 2722fdc..c8d2886 100644
--- a/cpp/src/dave/frame_processors.cpp
+++ b/cpp/src/dave/frame_processors.cpp
@@ -24,6 +24,9 @@ std::pair<bool, size_t> OverflowAdd(size_t a, size_t b)
     bool didOverflow = _addcarry_u64(0, a, b, &res);
 #elif defined(_MSC_VER) && defined(_M_IX86)
     bool didOverflow = _addcarry_u32(0, a, b, &res);
+#elif defined(_MSC_VER)
+    res = a + b;
+    bool didOverflow = (res < a);
 #else
     bool didOverflow = __builtin_add_overflow(a, b, &res);
 #endif
